### ISSUE TITLE
Parse function stucks

### ISSUE THREAD

**Issue by SergeVil** [json4s/json4s#335]
Tried to parse such wrong Json suntax, but the function parse() hangs on. 

{ueNm""eeam2""ulce""IfAGSGI3QBQA4NDBQBQEj0085KHwYiHsGEBxDVf51sxOQFFj75HNvYHLOYRk5Tto7DISlX0/BTneFfytio0xxeXwJABtq/aa/m+fShVHQDQB,raOlCukSS:sgJyC52UqKhaDtYhScm3PoJCM3&t21-12T934%A1&e22-12T934%A1&v21-40&prs=""rtOlCagsA""i=jJc0ZkaEi1mJ8Mej5InOl%Ds=060-91%A832Zs=060-91%A832Zs=050-5s=&rc,ealdrs""eeaaioagalcm}

I would expect the exception.

I use org.json4s:json4s-native_2.11:3.3.0

**Comment by jschlather** [comment]
This appears to be fixed on the 3.4 branch.

```scala> import org.json4s.native.JsonMethods._
import org.json4s.native.JsonMethods._

scala> val str = """{ueNm""eeam2""ulce""IfAGSGI3QBQA4NDBQBQEj0085KHwYiHsGEBxDVf51sxOQFFj75HNvYHLOYRk5Tto7DISlX0/BTneFfytio0xxeXwJABtq/aa/m+fShVHQDQB,raOlCukSS:sgJyC52UqKhaDtYhScm3PoJCM3&t21-12T934%A1&e22-12T934%A1&v21-40&prs=""rtOlCagsA""i=jJc0ZkaEi1mJ8Mej5InOl%Ds=060-91%A832Zs=060-91%A832Zs=050-5s=&rc,ealdrs""eeaaioagalcm}"""
str: String = {ueNm""eeam2""ulce""IfAGSGI3QBQA4NDBQBQEj0085KHwYiHsGEBxDVf51sxOQFFj75HNvYHLOYRk5Tto7DISlX0/BTneFfytio0xxeXwJABtq/aa/m+fShVHQDQB,raOlCukSS:sgJyC52UqKhaDtYhScm3PoJCM3&t21-12T934%A1&e22-12T934%A1&v21-40&prs=""rtOlCagsA""i=jJc0ZkaEi1mJ8Mej5InOl%Ds=060-91%A832Zs=060-91%A832Zs=050-5s=&rc,ealdrs""eeaaioagalcm}

scala> parse(str)
org.json4s.ParserUtil$ParseException: unknown token u
Near: {u
  at org.json4s.native.JsonParser$Parser.fail(JsonParser.scala:234)
  at org.json4s.native.JsonParser$Parser.nextToken(JsonParser.scala:322)
  at org.json4s.native.JsonParser$$anonfun$1.apply(JsonParser.scala:185)
  at org.json4s.native.JsonParser$$anonfun$1.apply(JsonParser.scala:143)
  at org.json4s.native.JsonParser$.parse(JsonParser.scala:131)
  at org.json4s.native.JsonParser$.parse(JsonParser.scala:71)
  at org.json4s.native.JsonParser$.parse(JsonParser.scala:66)
  at org.json4s.native.JsonMethods$class.parse(JsonMethods.scala:11)
  at org.json4s.native.JsonMethods$.parse(JsonMethods.scala:63)
  ... 43 elided```

**Comment by xuwei-k** [comment]
http://scastie.org/20160 

Could not reproduce the problem.
